src/exporter/DaemonMetricCollector.cc

```diff
@@ -121,42 +121,50 @@ void DaemonMetricCollector::dump_asok_metrics() {
     for (auto &perf_group_item : counter_schema) {
       std::string perf_group = {perf_group_item.key().begin(),
                                 perf_group_item.key().end()};
-      json_object perf_group_object = perf_group_item.value().as_object();
-      auto counters = perf_group_object["counters"].as_object();
-      auto counters_labels = perf_group_object["labels"].as_object();
-      auto counters_values =
-          counter_dump[perf_group].as_object()["counters"].as_object();
-      labels_t labels;
-
-      for(auto &label: counters_labels) {
-        std::string label_key = {label.key().begin(), label.key().end()};
-        labels[label_key] = quote(label.value().as_string().c_str());
+      json_array perf_group_schema_array = perf_group_item.value().as_array();
+      json_array perf_grp_dump_array = counter_dump[perf_group].as_array();
+      for (auto schema_itr = perf_group_schema_array.begin(),
+                dump_itr = perf_grp_dump_array.begin();
+                schema_itr != perf_group_schema_array.end() &&
+                dump_itr != perf_grp_dump_array.end();
+                ++schema_itr, ++dump_itr) {
+                  auto& perf_schema_grp = *schema_itr;
+                  auto& perf_dump_grp = *dump_itr;
```

Comment:
BTW I'm not sure you want to use `at()` here since it throws if the specified key is not there.  Of course `"labels"` and `"counters"` are expected to always be there, but ceph-exporter should be robust and not crash if invalid input is encountered.

Reply:
Thanks @idryomov for suggestion. Somehow my IDE is changing indentation,  addressed the changes suggested. 
I'm not sure how else I could get  `counters` and `labels` object from a json_array. It shouldn't be an issue as long as counter dump format stays the same.

Reply:
> I'm not sure how else I could get `counters` and `labels` object from a json_array. It shouldn't be an issue as long as counter dump format stays the same.

Yeah, and it looks like `as_array()`, `as_object()`, etc throw too so the issue is more widespread.